src/Controller/user/saveUserData.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  NotFoundResponseSchema,
+  ServerErrorResponseSchema,
+  UnAuthorizedResponseSchema,
+} from "../../Schemas/error.schema";
+
+import { FastifyInstance, FastifyReply, FastifyRequest } from "fastify";
+// import { Type, Static } from "@sinclair/typebox";
+import UserDB from "../../Models/UserModel";
+
+// const BodySchema = Type.Strict(
+//   Type.Object({
+//     name: Type.String({ minLength: 1 }),
+//     email: Type.String({ format: "email" }),
+//     age: Type.String({ minLength: 1 }),
+//   })
+// );
+
+// type BodyType = Static<typeof BodySchema>;
+
+export default function createUserRoute(fastify: FastifyInstance) {
+  fastify.post(
+    "/",
+    {
+      schema: {
+        response: {
+          "401": UnAuthorizedResponseSchema,
+          "404": NotFoundResponseSchema,
+          "500": ServerErrorResponseSchema,
+        },
+      },
+    },
+    async (req: FastifyRequest, rep: FastifyReply) => {
+      const userData = req.user;
+
+      console.log(userData?.emailAddresses);
+
+      try {
+        const userExist = await UserDB.findOne({
```

Comment:
Create a **Services** folder and a **Repository** folder. Follow the same structure as your Controller space. Move the logical block to services and the DB accessor functions to repositories. The Controller should be thin, and the services should handle the logical part.

Reply:
sure i will do it right away !!